python/TestHarness/testers/RunApp.py

```diff
@@ -27,6 +27,7 @@ def validParams():
         params.addParam('executable_pattern', "A test that only runs if the exectuable name matches the given pattern")
         params.addParam('delete_output_before_running',  True, "Delete pre-existing output files before running test. Only set to False if you know what you're doing!")
         params.addParam('delete_output_folders', True, "Delete output folders before running")
+        params.addParam('max_screen_output_size', 100000, "Bytes allowed in stdout/stderr before it is subjected to being trimmed. Set to -1 to ignore output size restrictions")
```

Comment:
Couldn't think of a better self describing param name.
max_output_buffer?

Maybe it should even be min_*?, since its the min character count before triggering threshold.